apps/ejabberd/src/ejabberd_c2s.erl

```diff
@@ -1949,7 +1965,10 @@ privacy_check_packet(StateData, From, To, Packet, Dir) ->
                        Packet :: jlib:xmlel(),
                        Dir :: 'in' | 'out') -> boolean().
 is_privacy_allow(StateData, From, To, Packet, Dir) ->
-    allow == privacy_check_packet(StateData, From, To, Packet, Dir).
+    case privacy_check_packet(StateData, From, To, Packet, Dir) of
+        allow -> true;
+        _ -> deny
```

Comment:
If we change `false` to `deny` then maybe we should also change `true` to `allow`?